File: rest/src/main/java/io/sparkled/rest/service/renderpreview/RenderPreviewRestService.java
package io.sparkled.rest.service.renderpreview;

import io.sparkled.model.entity.SequenceChannel;

import javax.inject.Inject;
import javax.ws.rs.*;
import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;
import java.util.List;

@Path("/renderPreview")
public class RenderPreviewRestService {

    private final RenderPreviewRestServiceHandler handler;

    @Inject
    public RenderPreviewRestService(RenderPreviewRestServiceHandler handler) {
        this.handler = handler;
    }

    @POST
    @Consumes(MediaType.APPLICATION_JSON)
    @Produces(MediaType.APPLICATION_JSON)
    public Response getRenderedSequence(@QueryParam("startFrame") int startFrame,
                                        @QueryParam("durationFrames") int durationFrames,
                                        List<SequenceChannel> sequenceChannels) {
        return handler.getRenderedSequence(startFrame, durationFrames, sequenceChannels);
    }
}

File: rest/src/main/kotlin/io/sparkled/rest/service/renderpreview/RenderPreviewRestService.kt
package io.sparkled.rest.service.renderpreview

import io.sparkled.model.entity.SequenceChannel
import javax.inject.Inject
import javax.ws.rs.Consumes
import javax.ws.rs.POST
import javax.ws.rs.Path
import javax.ws.rs.Produces
import javax.ws.rs.QueryParam
import javax.ws.rs.core.MediaType
import javax.ws.rs.core.Response

@Path("/renderPreview")
class RenderPreviewRestService @Inject
constructor(private val handler: RenderPreviewRestServiceHandler) {

    @POST
    @Consumes(MediaType.APPLICATION_JSON)
    @Produces(MediaType.APPLICATION_JSON)
    fun getRenderedSequence(
        @QueryParam("startFrame") startFrame: Int,
        @QueryParam("frameCount") frameCount: Int,
        sequenceChannels: List<SequenceChannel>
    ): Response {
        return handler.getRenderedSequence(startFrame, frameCount, sequenceChannels)
    }
}